### PR TITLE
Support detailed error codes

### PR DESCRIPTION
**Pull request by weichihl**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/source/app/kvsapp.c [hunk 1]
```diff
@@ -513,38 +517,56 @@ static int updateAndVerifyRestfulReqParameters(KvsApp_t *pKvs)
 
 static int setupDataEndpoint(KvsApp_t *pKvs)
 {
-    int res = ERRNO_NONE;
+    int xRes = KVS_ERRNO_NONE;
```
Comment (codingspirit) [inline src/source/app/kvsapp.c:520]: Not critical but I think `x` prefix was used as non stdint types.

File: doc/kvs_error_code.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# KVS Error Code
```
Comment (codingspirit) [inline doc/kvs_error_code.md:1]: How about create a chapter in main `README.md` about `Debug and issue address` link to this?

**Review by codingspirit — APPROVED** [review]
This commit is too big to review -_-|
Few suggestions are attached but not mandatory to fix

File: doc/kvs_error_code.md (hunk 2 above)
Comment (weichihl) [reply doc/kvs_error_code.md:1]: Yeah, that's another TODO task to improve the main `README.md`. For now, I'm just adding a little doc here.

File: src/source/app/kvsapp.c (hunk 1 above)
Comment (weichihl) [reply src/source/app/kvsapp.c:520]: Yes, you are right. I'll make the change in this PR since this is a big PR.

**Pull request merged**